devito/types/basic.py

```diff
@@ -341,10 +341,16 @@ def _filter_assumptions(cls, **kwargs):
         # pop predefined assumptions
         for key in ('real', 'imaginary', 'commutative'):
             kwargs.pop(key, None)
+
         # extract sympy.Symbol-specific kwargs
         for i in list(kwargs):
             if i in _assume_rules.defined_facts:
                 assumptions[i] = kwargs.pop(i)
+
+        # Handle the nonnegative argument
+        if 'nonnegative' in assumptions and assumptions['nonnegative'] is None:
```

Comment:
that's just `assumptions.pop('nonnegative', None)`

Reply:
No it's not. If `nonnegative` is provided as a kwarg, but is not `None` then you don't want to pop it.